modules/caddyhttp/reverseproxy/streaming.go

```diff
@@ -32,21 +32,20 @@ func (h Handler) handleUpgradeResponse(rw http.ResponseWriter, req *http.Request
 	reqUpType := upgradeType(req.Header)
 	resUpType := upgradeType(res.Header)
 	if reqUpType != resUpType {
-		// TODO: figure out our own error handling
-		// p.getErrorHandler()(rw, req, fmt.Errorf("backend tried to switch protocol %q when %q was requested", resUpType, reqUpType))
+		h.logger.Sugar().Errorf("backend tried to switch protocol %q when %q was requested", resUpType, reqUpType)
 		return
 	}
 
 	copyHeader(res.Header, rw.Header())
 
 	hj, ok := rw.(http.Hijacker)
 	if !ok {
-		// p.getErrorHandler()(rw, req, fmt.Errorf("can't switch protocols using non-Hijacker ResponseWriter type %T", rw))
+		h.logger.Sugar().Errorf("can't switch protocols using non-Hijacker ResponseWriter type %T", rw)
```

Comment:
Can be ERROR level structured log

Reply:
When you say structured log, do you mean throw away the `%T` bit of the error? Or how do I encode that with zap? I don't think there's a `zap.TypeOf` or whatever... 🤔

Reply:
I mean use `h.Logger.Error(...)` :)